src/components/loading/LoadingEmitter.ts

```diff
@@ -0,0 +1,5 @@
+import EventEmitter from "events";
+
+const loadingEmitter = new EventEmitter();
+
+export default loadingEmitter;
```

Comment:
Prefer not export this emitter directly but export function that interact with it. Therefore you can enforce event type and function type for the on and off function.